### ISSUE TITLE
How to Write Input_Signature's for function's and model's names when using tf.fucntion?

### ISSUE THREAD

**Issue by yourtheron** [tensorflow/tensorflow#35487]
Hi,

I want to decorate tf.function unto my training function
```
@tf.function
def trainOneSample(data, model, optimizer, lossFunc, accFunc):
...
```
where data are a pair of 5D arrays with float32 and uint8 types (for input and label), model is the name of my NN model, optimiser is the name of an optimisation object, lossFunc and accFunc are names of custom functions. How can I decorate them in tf.function by Input_Signature? I cannot find out any solution in tf's tutorial or guide.

**Comment by gowthamkpr** [comment]
@yourtheron Please take a look at this [issue](https://github.com/tensorflow/tensorflow/issues/28165) and let me know if it helps. Thanks!

**Comment by yourtheron** [comment]
@gowthamkpr Got it. So it seems that I do not have to provide any input signature for optimizer, lossFunc, or accFunc here when defining and decorating trainOneSample(...), is that correct? 
A by-product question: My trainOneSample(...) function will execute my model's call(...) method, and of course, other operations such as lossFunc(...) and accFunc(...). So in order to let tf create a graph and concrete functions, shall I decorate tf.function unto all of those functions as well as provide input signatures to each one of them or decorate trainOneSample(...)  only? All of inputs to those functions are tf tensors.

**Comment by gowthamkpr** [comment]
@yourtheron Look at this specific issue on how they used `@tf.function` This should give you clarity on how to use it in your case.

**Comment by gowthamkpr** [comment]
@yourtheron Can I close this issue as it is not a bug. Thanks!

**Comment by yourtheron** [comment]
@gowthamkpr Sorry for the late reply. Sure, thanks for your help!
